### PR TITLE
[fix](datetimev2) Fix BE datetimev2 type returning wrong result

### LINKED ISSUE #15813: [Bug] datetimev2 type returning wrong result when developing group_uniq_array function
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?

#15339 
When @TangSiyang2001 was developing `group_uniq_array` function and doing regression test, he found a `datetimev2` bug.
1. creat a test table.
```sql
create table test (
    c_id INT,
    c_date DATE,
    c_datev2 DATEV2,
    c_date_time DATETIME,
    c_date_timev2 DATETIMEV2(6))
         DISTRIBUTED BY HASH(c_id) BUCKETS 3
         PROPERTIES (
                "replication_num" = "1"
         ) ;
Query OK, 0 rows affected (0.05 sec)
```
2. insert data.
```sql
insert into test values(
1, 
'2022-12-01', 
'2022-12-01', 
'2022-12-01 22:23:23', 
'2022-12-01 22:23:24.999999');
Query OK, 1 row affected (0.14 sec)
{'label':'insert_b85de801adae47db_a6f1d6297468dbfb', 'status':'VISIBLE', 'txnId':'1006'}

insert into test values(
2, 
'2022-12-02', 
'2022-12-02', 
'2022-12-02 22:23:23', 
'2022-12-02 22:23:24.999999');
Query OK, 1 row affected (0.06 sec)
{'label':'insert_8b8fd04b50674ffe_9ff5e5c2955abf50', 'status':'VISIBLE', 'txnId':'1007'}
```
3. select from test table.
```sql
select * from test;
+------+------------+------------+---------------------+----------------------------+
| c_id | c_date     | c_datev2   | c_date_time         | c_date_timev2              |
+------+------------+------------+---------------------+----------------------------+
|    2 | 2022-12-02 | 2022-12-02 | 2022-12-02 22:23:23 | 2022-12-02 22:23:24.999999 |
|    1 | 2022-12-01 | 2022-12-01 | 2022-12-01 22:23:23 | 2022-12-01 22:23:24.999999 |
+------+------------+------------+---------------------+----------------------------+
2 rows in set (0.04 sec)

select
            group_uniq_array(c_date),
            group_uniq_array(c_datev2),
            group_uniq_array(c_date_time),
            group_uniq_array(c_date_timev2)
from test;
+----------------------------+------------------------------+--------------------------------------------+-----------------------------------+
| group_uniq_array(`c_date`) | group_uniq_array(`c_datev2`) | group_uniq_array(`c_date_time`)            | group_uniq_array(`c_date_timev2`) |
+----------------------------+------------------------------+--------------------------------------------+-----------------------------------+
| [2022-12-02, 2022-12-01]   | [2022-12-01, 2022-12-02]     | [2022-12-02 22:23:23, 2022-12-01 22:23:23] | [, ]                              |
+----------------------------+------------------------------+--------------------------------------------+-----------------------------------+
1 row in set (0.09 sec)
```

We have tested functions `add`, `merge`, `insert_result_into` in `aggregate_function_group_uniq_array.h` return the right results for `datetimev2`.

### What You Expected?

```sql
select
            group_uniq_array(c_date),
            group_uniq_array(c_datev2),
            group_uniq_array(c_date_time),
            group_uniq_array(c_date_timev2)
from test;
+----------------------------+------------------------------+--------------------------------------------+-----------------------------------+
| group_uniq_array(`c_date`) | group_uniq_array(`c_datev2`) | group_uniq_array(`c_date_time`)            | group_uniq_array(`c_date_timev2`) |
+----------------------------+------------------------------+--------------------------------------------+-----------------------------------+
| [2022-12-02, 2022-12-01]   | [2022-12-01, 2022-12-02]     | [2022-12-02 22:23:23, 2022-12-01 22:23:23] | [2022-12-02 22:23:24.999999, 2022-12-01 22:23:24.999999]                              |
+----------------------------+------------------------------+--------------------------------------------+-----------------------------------+
```

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Yukang-Lian**
# Proposed changes

Issue Number: close #15813 

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 36.96 seconds
 load time: 492 seconds
 storage size: 17123088829 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230112194446_clickbench_pr_79184.html

File: be/src/vec/sink/vmysql_result_writer.cpp [hunk 1]
```diff
@@ -367,6 +367,14 @@ int VMysqlResultWriter::_add_one_cell(const ColumnPtr& column_ptr, size_t row_id
         char buf[64];
         char* pos = datev2.to_string(buf);
         return buffer.push_string(buf, pos - buf - 1);
+    } else if (which.is_date_time_v2()) {
+        auto& column_vector = assert_cast<const ColumnVector<UInt64>&>(*column);
+        auto value = column_vector[row_idx].get<UInt64>();
+        DateV2Value<DateTimeV2ValueType> datetimev2;
```
Comment (Gabriel39) [inline be/src/vec/sink/vmysql_result_writer.cpp:373]: Use `DateV2Value<DateTimeV2ValueType> datetimev2 = binary_cast<UInt64, DateV2Value<DateTimeV2ValueType>>(value)`.

Could you also change datev2 to this method ?

**Comment by Gabriel39** [comment]
Please add a regression case

File: be/src/vec/sink/vmysql_result_writer.cpp (hunk 1 above)
Comment (Yukang-Lian) [reply be/src/vec/sink/vmysql_result_writer.cpp:373]: sure, will do.

**Review by Gabriel39 — APPROVED** [review]
LGTM

**Pull request merged**